server/lib/chat/chat-service.ts

```diff
@@ -406,6 +400,43 @@ export default class ChatService {
     return getUsersForChannel(originalChannelName)
   }
 
+  async getChatUserProfile(channelName: string, userId: SbUserId, targetId: SbUserId) {
+    const userSockets = this.getUserSockets(userId)
+    const originalChannelName = await this.getOriginalChannelName(channelName)
+    if (
+      !this.state.users.has(userSockets.userId) ||
+      !this.state.users.get(userSockets.userId)!.has(originalChannelName)
+    ) {
+      throw new ChatServiceError(
+        ChatServiceErrorCode.InvalidGetUsersAction,
```

Comment:
I find this error code naming to be pretty bad, it's basically "generic error for route X". This particular usage would be better named "NotInChannel" or something